src/pivot-table/data/__tests__/use-pv-data.test.ts

```diff
@@ -116,21 +147,19 @@ describe("useData", () => {
   });
 
   test("should return data", () => {
-    const { result } = renderHook(() => useData(qPivotDataPages, layoutService));
+    const result = usePVData({ qPivotDataPages, layoutService, pageInfo });
 
-    expect(result.current.headersData).toBe(headersData);
-    expect(result.current.measureData).toBe(measureData);
-    expect(result.current.topDimensionData).toBe(topDimensionData);
-    expect(result.current.leftDimensionData).toBe(leftDimensionData);
+    expect(result.headersData).toBe(headersData);
+    expect(result.measureData).toBe(measureData);
+    expect(result.topDimensionData).toBe(topDimensionData);
+    expect(result.leftDimensionData).toBe(leftDimensionData);
   });
 
-  test("calling nextPageHandler should trigger data updates", () => {
-    const { result } = renderHook(() => useData(qPivotDataPages, layoutService));
+  test.skip("calling nextPageHandler should trigger data updates", () => {
```

Comment:
in this test, what is happening is that `result.nextPageHandler(...)`should trigger a state change. Based on that state change -> an `useEffect` might be called, and we are asserting the function calls within that `useEffect` (which are mostly mocked), but since we are mocking the `useEffect` itself (just calling the callback that is being passed into it, not recall that callback in every dependency check) this becomes impossible to handle || at least i don't have any idea to how to handle it. 

Any suggestions are welcome on this one 🙏